### PR TITLE
Updates dokku-daemon for compatibility

### PR DESCRIPTION
**Pull request by dean1012**
* Allow use of dokku-daemon with newer dokku versions
* Updates documentation

### CODE REVIEW

**Review by josegonzalez — APPROVED** [review]

**Comment by josegonzalez** [comment]
I'll merge once tests pass.

**Comment by josegonzalez** [comment]
Just be sure to comment here when they do to remind me :)

**Comment by dean1012** [comment]
@josegonzalez I am running low on time at the moment and don't have time to look at this right now. I'll get to it when I can but hopefully, someone else in the community will find time before I do.

I can confirm that this works for me when using the following:

* dokku 0.18.3
* ubuntu bionic
* this pull request

**Comment by dean1012** [comment]
@josegonzalez Just wanted to update you...

I've got all the testing fixed except one thing. The `client_command` function uses `nc` to pipe commands into the socket and get the result.

Unfortunately, it seems that `nc` is not sending the resulting output anywhere that I can find which causes all the tests relying on that function to fail.

I strongly believe that once this function is fixed, all tests will work.

I am continually looking into this but would appreciate any insight from the community as well.

Look forward to merging this.

Also, an unrelated tidbit... I ran into many problems with the `dokku-api` software not properly handling JSON returned by `dokku-daemon` for the `logs` command. I ended up making my own software based on `dokku-api` with a much simpler feature set and based on NodeJS. It is hosted at https://github.com/dean1012/dokku-api-nodejs

**Comment by dean1012** [comment]
@josegonzalez Would you look at that? All tests pass!

To recap, this pull request essentially does the following:

* Updates to latest version of Dokku and Ubuntu Bionic
* Updates tests to work better against later version of Dokku
* Replaces `nc` with a new NodeJS test program I wrote (source included alongside pre-compiled small binary under tests directory). This allows tests to actually run successfully now. Something along the way broke with `nc` and `socat` which prevents input/output with the socket from working properly. This interferes with tests. I think what I did will be acceptable here.
* Updates Dokku daemon service files to work more reliably with a larger proportion of operating systems
* Updates Dokku daemon to escape double quotes in addition to new lines. This allows commands like `logs myapp` to work whereas before they would generate an error. Pull request #18 can be closed as a duplicate of this one as it's featureset is included here.
* Fixes a few misc. issues preventing TravisCI from working and/or passing (such as GPG keys being out of date)

Please merge when able! I am happy I was able to help this project out.

**Comment by dean1012** [comment]
Added a last minute fix to a corner case bug. Should be good now once the tests pass.

File: bin/dokku-daemon
Comment (josegonzalez) [inline bin/dokku-daemon:75]: Can you explain this?

File: init/dokku-daemon.service
Comment (josegonzalez) [inline init/dokku-daemon.service:13]: Why these removals?

File: bin/dokku-daemon
Comment (josegonzalez) [inline bin/dokku-daemon:132]: This removes stdin/stdout?

File: tests/functional.bats
Comment (josegonzalez) [inline tests/functional.bats]: `app` is also not a valid command, was this necessary?

File: bin/dokku-daemon
Comment (dean1012) [reply bin/dokku-daemon:75]: At least in the environments I tested in, I found that later versions of SystemD had issues with permissions on the created socket. By adding the permissions to `socat` directly, I was able to resolve the issue. I wasn't able to identify any negative consequences to this.

File: tests/functional.bats
Comment (dean1012) [reply tests/functional.bats]: i think this one can be chalked up to a bout of confusion on my part late at night. i'll revert this and verify tests pass.

File: bin/dokku-daemon
Comment (dean1012) [reply bin/dokku-daemon:132]: I had to make many changes and pull requests recently related to dokku and letsencrypt both, so I cannot remember the reasoning behind this change.

I will revert it and test to verify functionality.

File: init/dokku-daemon.service
Comment (dean1012) [reply init/dokku-daemon.service:13]: I had to make many changes and pull requests recently related to dokku and letsencrypt both, so I cannot remember the reasoning behind this change.

I will revert it and test to verify functionality.

**Comment by dean1012** [comment]
Give me a little bit of time to make a new commit with requested changes and do some testing.

**Comment by dean1012** [comment]
All passing and I believe I've addressed all your concerns now @josegonzalez - you can merge now.

**Comment by dean1012** [comment]
@josegonzalez is there anything else you need me to do here before you can merge? Just checking, thanks!

**Comment by dean1012** [comment]
Can we please merge this request @josegonzalez ? Thank you!

**Pull request merged**
